A pure virtual function is not implemented

```
Compiling libretro_os.cpp...
../../../../backends/platform/libretro/libretro_os.cpp: In function ‘OSystem* retroBuildOS(bool)’:
../../../../backends/platform/libretro/libretro_os.cpp:1327:45: error: invalid new-expression of abstract class type ‘OSystem_RETRO’
 1327 |    return new OSystem_RETRO(aEnableSpeedHack);
      |                                             ^
../../../../backends/platform/libretro/libretro_os.cpp:345:7: note:   because the following virtual functions are pure within ‘OSystem_RETRO’:
  345 | class OSystem_RETRO : public EventsBaseBackend, public PaletteManager {
      |       ^~~~~~~~~~~~~
In file included from ../../../../backends/base-backend.h:25,
                 from ../../../../backends/platform/libretro/libretro_os.cpp:33:
../../../../common/system.h:1190:22: note:     ‘virtual void OSystem::showOverlay(bool)’
 1190 |         virtual void showOverlay(bool inGUI = true) = 0;
      |                      ^~~~~~~~~~~
make: *** [Makefile:577: ../../../../backends/platform/libretro/libretro_os.o] Error 1

```

Scummvm master added a new default argument to these functions 3 days ago:

https://github.com/scummvm/scummvm/commit/87bad2cc7d602488ba3fbdf8ff067d3d74fc2b85


not sure sure if it's the only error.

--- a/libretro_os.cpp
+++ b/libretro_os.cpp
@@ -351,6 +351,7 @@ class OSystem_RETRO : public EventsBaseBackend, public PaletteManager {
 
       Graphics::Surface _overlay;
       bool _overlayVisible;
+      bool _overlayInGUI;
 
       Graphics::Surface _mouseImage;
       RetroPalette _mousePalette;
@@ -535,7 +536,7 @@ class OSystem_RETRO : public EventsBaseBackend, public PaletteManager {
 
       virtual void updateScreen()
       {
-         const Graphics::Surface& srcSurface = (_overlayVisible) ? _overlay : _gameScreen;
+         const Graphics::Surface& srcSurface = (_overlayInGUI) ? _overlay : _gameScreen;
          if(srcSurface.w && srcSurface.h)
          {
             switch(srcSurface.format.bytesPerPixel)
@@ -590,14 +591,16 @@ class OSystem_RETRO : public EventsBaseBackend, public PaletteManager {
          // TODO
       }
 
-      virtual void showOverlay()
+      virtual void showOverlay(bool inGUI)
       {
-         _overlayVisible = true;
+        _overlayVisible = true;
+        _overlayInGUI = inGUI;
       }
 
       virtual void hideOverlay()
       {
          _overlayVisible = false;
+         _overlayInGUI = false;
       }
 
       virtual void clearOverlay()
@@ -840,7 +843,7 @@ class OSystem_RETRO : public EventsBaseBackend, public PaletteManager {
 
       const Graphics::Surface& getScreen()
       {
-         const Graphics::Surface& srcSurface = (_overlayVisible) ? _overlay : _gameScreen;
+         const Graphics::Surface& srcSurface = (_overlayInGUI) ? _overlay : _gameScreen;
 
          if(srcSurface.w != _screen.w || srcSurface.h != _screen.h)
          {

